source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,807 @@
+ * @param[in|out] peEMACState: the current state in the call to xNetworkInterfaceInitialise.
+ *                   The state will be updated to the next state depending on the status of
+ *                   the PHY initialization.
+ *
+ * @return kStatus_Success if the PHY was initialized; error code otherwise.
+ */
+static status_t xSetupPHY( phy_config_t * pxConfig,
+                           EMACState_t * peEMACState )
+{
+    status_t xStatus;
+
+    /* Make sure that current state is correct. */
+    configASSERT( *peEMACState == xEMAC_SetupPHY );
+
+    ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
+    ethernetifLocal->TxBuffDescrip = &( txBuffDescrip_0[ 0 ] );
+    ethernetifLocal->RxDataBuff = &( rxDataBuff_0[ 0 ] );
+    ethernetifLocal->TxDataBuff = &( txDataBuff_0[ 0 ] );
+    ethernetifLocal->base = ( void * ) ENET_BASE;
+
+    /* Set the clock frequency. */
+    mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+    mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+    /* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+    pxConfig->autoNeg = pdTRUE;
+    pxConfig->duplex = kPHY_FullDuplex;
+    pxConfig->phyAddr = BOARD_ENET0_PHY_ADDRESS;
+    pxConfig->speed = kPHY_Speed100M;
+
+    FreeRTOS_printf( ( "Starting PHY initialization.\r\n" ) );
+
+    xStatus = PHY_Init( &phyHandle, pxConfig );
+
+    if( xStatus != kStatus_Success )
+    {
+        FreeRTOS_printf( ( "Failed to initialize the PHY..." ) );
+        *peEMACState = xEMAC_Fatal;
+    }
+    else
+    {
+        *peEMACState = xEMAC_WaitPHY;
+    }
+
+    return xStatus;
+}
+
+/*
+ * @brief This function is used wait on the auto-negotiation completion. In case auto-negotiation
+ *        fails, the function tries to use the default values (100M speed; full duplex communication
+ *        link) to get the link up.
+ *
+ * @param[out] pxConfig: the updated configuration parameters will be written into this pointer.
+ * @param[in|out] peEMACState: the current state in the call to xNetworkInterfaceInitialise.
+ *                   The state will be updated to the next state depending on the status of
+ *                   the PHY initialization.
+ *
+ * @return kStatus_Success if the PHY was initialized; error code otherwise.
+ */
+static status_t xWaitPHY( phy_config_t * pxConfig,
+                          EMACState_t * peEMACState )
+{
+    status_t xStatus;
+    bool LinkUp;
+    bool autoNegotiationComplete;
+    uint8_t ucCounter = 0;
+
+    /* Make sure that current state is correct. */
+    configASSERT( *peEMACState == xEMAC_WaitPHY );
+
+    do
+    {
+        xStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+        if( autoNegotiationComplete == true )
+        {
+            break;
+        }
+
+        /* Try for only a limited number of times. */
+        if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
```

Comment:
MAX_AUTONEG_FAILURE_COUNT is 5, this runs 5 time.. So auto-negotiation times out it 50ms?

Reply:
FIXED